packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/totp_setup_state_machine.dart

```diff
@@ -59,29 +62,59 @@ final class TotpSetupStateMachine
         )
         .result;
     _session = response.session;
+    _details = TotpSetupDetails(
+      username: CognitoIdToken(tokens.idToken).username,
+      sharedSecret: response.secretCode!,
+    );
     emit(
       TotpSetupState.requiresVerification(
-        TotpSetupDetails(
-          username: CognitoIdToken(tokens.idToken).username,
-          sharedSecret: response.secretCode!,
-        ),
+        _details!,
       ),
     );
   }
 
   Future<void> _onVerify(TotpSetupVerify event) async {
     final tokens = await manager.getUserPoolTokens();
     final accessToken = tokens.accessToken.raw;
-    await _cognitoIdp
-        .verifySoftwareToken(
-          VerifySoftwareTokenRequest(
-            accessToken: accessToken,
-            session: _session,
-            userCode: event.code,
-            friendlyDeviceName: event.friendlyDeviceName,
+    try {
+      await _cognitoIdp
+          .verifySoftwareToken(
+            VerifySoftwareTokenRequest(
+              accessToken: accessToken,
+              session: _session,
+              userCode: event.code,
+              friendlyDeviceName: event.friendlyDeviceName,
+            ),
+          )
+          .result;
+    } on Exception catch (e, st) {
+      // Handle mismatch code exception that may occur during TOTP verification.
+      // See: https://docs.aws.amazon.com/cognito-user-identity-pools/latest/APIReference/API_VerifySoftwareToken.html#API_VerifySoftwareToken_Errors
+      // TODO(equartey): Investigate when CodeMismatchException is thrown and handle accordingly.
```

Comment:
reading the exceptions' discription seems the `CodeMismatchException` is thrown when the totp is already setup and later when signing up the user using a wrong code. I'm not very familiar with our auth state machine so not sure if we have different flow/state machine for setting up the totp and then using totp code in the sign-in flow

Reply:
I was curious about this too, though I could not get that exception to throw. Our state machine blocks calls to `Amplify.Auth.verifyTotpSetup()` when the client is not in the setup process. After an authenticator app is setup, the totp code gets submitted during sign. As it's setup now, I don't think this is a concern.

Reply:
note: I omitted the TODO comment after reviewing it more.